### DISCUSSION TITLE
Add different nozzle sizes with profiles separated from standard 0.4

### DISCUSSION THREAD

**Discussion by rideorroot** [OrcaSlicer/OrcaSlicer#418] [Q&A]
Hey everyone!

I have set up multiple printer and printing profiles for my Prusa Mini since it has a Revo Micro installed and all profiles show up independently of the nozzle size and I'm looking for a way to restrict them to a particular printer profile.
I guess I'm looking for a feature like Prusa Slicers "Compatible printers".

Maybe adding Json files to the configuration folder will work?
I have no experience with that so I wanted to ask before messing something up.

**Comment by RagingRoosevelt** [comment]
Or even just a way to have an alternative nozzle size.  I have a Voron Trident but I run it at 0.6mm, not 0.4mm and I haven't figured out how to add it with something besides 0.4mm.

**Reply by rideorroot** [reply]
You can simply edit the printer profile and change the nozzle size there.

<img width="1204" alt="Screen Shot 2023-03-04 at 11 13 20 AM" src="https://user-images.githubusercontent.com/89993948/222894178-b1a78637-9c3f-4117-8b4a-662eb13eefe6.png">

It's fine imo if you only use one nozzle size but once you have one printer with a quick swap system and want to add different nozzle sizes it get's really messy because all profiles and filaments are listed.

**Reply by Mario-vminion** [reply]
is it only the nozzle size you have to change or do you have to adjust the filament profile also

**Comment by CodyPrince** [comment]
Superslicer and Prusaslicer call them dependencies. I posted question earlier today, but just now found this request. A must have if your using different nozzles.

![Capture](https://github.com/SoftFever/OrcaSlicer/assets/124317309/7bbad9aa-6696-42e0-b8ff-b3b8bef48baa)

**Comment by Hunkoys** [comment]
Yeah. This. Especially when using a lot of filaments with different settings(ie. PA). Show only filament profiles for a certain nozzle size. 

1. Create a printer profile base on a base printer. 
2. Any new filament created while in the newly created printer will only belong and show when said printer is selected.
3. Same with newly created print profile

Also, this way, we can have a sequencial generated id or uuid instead of using the name as id's. I know this would change a lot of the current infrastructure but it will add QoA. Not having colliding print profile names from other printers would be nice. Having @printername at the end of profiles could be distracting.

**Reply by Kosygor** [reply]
Hello I came to 3D printing lately and today switched nozzles for the first time in my life. Functionality described here seems so essential and obvious that I thought my orca slicer was messed up. If you play around with one nozzle and few filaments it's okay but if you start playing with different nozzles and tens of filaments it is hard to get your head around it.

**Reply by raunchyneworleans** [reply]
[Kosygor](https://github.com/Kosygor) if you change the nozzle size you also need to change the with of the layer, the layer hight. maybe lower or raise the temp a bit. adjust speeds. pressure advance changes. and so on, so yeah that's why i like to have the option to select a different nozzle size in a menu. and then most changes are already done. because like 90% of the changes are always the same. add the changes you have to do with changing from abs to pla to petg to carbon to glow in the dark and so on it really helps to just select 0.4 or 0.6/0.2 and adjust a little for your personal taste and machine. that the also realy helps people like you. but most used is 0.4 atm then 0.6 and 0.2 0.6 because it is way faster then 0.4 but you van have almost the same level of detail. and 0.2 is for people who want to print those figures and other stuff that realy needs the detail and is not to big. so in basics smaller is more detail and slower and bigger is faster and stronger. i limited myself in the beginning to 0.4 and only printed a few thinks in 0.6 and even less in 0.2. the material the nozzles are made of don't realy matter in the beginning. just remember that more abrasive material can wear the nozzle faster so you have options for harder nozzles to counteract that wear. that is all the information you need about the nozzles. hope it helped and ask me anything if you need more explanation in simple terms. i guess you where just overwhelmed because of all the options and that almost everyone in this hobby only does deep dives 😅 sorry about that.

**Comment by CelsoSantos** [comment]
In my case, I have Revo Six nozzles and would very much like the ability to change profiles depending on the selected nozzle and/or filament

**Comment by raunchyneworleans** [comment]
yeah. need multiple nozzle sizes myself 2. most other profiles already have them. i would think having one up and one down from 0.4mm would be kinda standard for all profiles these days.

**Comment by Mabe75** [comment]
If you think about where the slicer originates from, you realise why these issues exist. The Bambu Lab serves a defined number of printers with certain properties. These are stored in the profiles or printer models.

However, if you now have a customised printer or an exotic printer, you will encounter these problems.

- If you change the nozzle size in the extruder area, the line widths are not adjusted. 
-It is a very tedious job to adjust all values to the nozzle width, with the risk of overlooking a value.
- Useful specifications that exist for 0.4 have to be figured out manually for the 0.6 (or other sizes). 
- I have also noticed that the maximum speed in the extruder section of the printer is not taken over. 

I wonder if I am doing something fundamentally wrong or if it is actually implemented that way. If it's not an error on my side, the slicer is unfortunately useless for me, as I don't have any printers with a ready-made profile
